sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/graph/elements.scala

```diff
@@ -0,0 +1,270 @@
+    throw new IllegalStateException(
+      "Cannot identify whether the table is streaming table or not. You may need to resolve the " +
+      "flows into table."
+    )
+  }
+
+  /**
+   * Get the DatasetType of the table
+   */
+  def datasetType: DatasetType = {
+    if (isStreamingTable) {
+      DatasetType.STREAMING_TABLE
+    } else {
+      DatasetType.MATERIALIZED_VIEW
+    }
+  }
+}
+
+/**
+ * A type of [[TableInput]] that returns data from a specified schema or from the inferred
+ * [[Flow]]s that write to the table.
+ */
+case class VirtualTableInput(
+    identifier: TableIdentifier,
+    specifiedSchema: Option[StructType],
+    incomingFlowIdentifiers: Set[TableIdentifier],
+    availableFlows: Seq[ResolvedFlow] = Nil
+) extends TableInput
+    with Logging {
+  override def origin: QueryOrigin = QueryOrigin()
+
+  assert(availableFlows.forall(_.destinationIdentifier == identifier))
+  override def load(readOptions: InputReadOptions): DataFrame = {
+    // Infer the schema for this virtual table
+    def getFinalSchema: StructType = {
+      specifiedSchema match {
+        // This is not a backing table, and we have a user-specified schema, so use it directly.
+        case Some(ss) => ss
+        // Otherwise infer the schema from a combination of the incoming flows and the
+        // user-specified schema, if provided.
+        case _ =>
+          SchemaInferenceUtils.inferSchemaFromFlows(availableFlows, specifiedSchema)
+      }
+    }
+
+    // create empty streaming/batch df based on input type.
+    def createEmptyDF(schema: StructType): DataFrame = readOptions match {
+      case _: StreamingReadOptions =>
+        MemoryStream[Row](ExpressionEncoder(schema, lenient = false), spark.sqlContext)
+          .toDF()
+      case _ => spark.createDataFrame(new util.ArrayList[Row](), schema)
+    }
+
+    val df = createEmptyDF(getFinalSchema)
+    df
+  }
+}
+
+/**
+ * Representing a view in the [[DataflowGraph]].
+ */
+trait View extends GraphElement {
+
+  /** Returns the unique identifier for this [[View]]. */
+  val identifier: TableIdentifier
+
+  /** Properties of this view */
+  val properties: Map[String, String]
+
+  /** User-specified comment that can be placed on the [[View]]. */
+  val comment: Option[String]
+}
+
+/**
+ * Representing a temporary [[View]] in a [[DataflowGraph]].
+ *
+ * @param identifier The identifier of this view within the graph.
+ * @param properties Properties of the view
+ * @param comment when defining a view
+ */
+case class TemporaryView(
+    identifier: TableIdentifier,
+    properties: Map[String, String],
+    comment: Option[String],
+    origin: QueryOrigin
+) extends View {}
+
+/**
+ * Representing a persisted [[View]] in a [[DataflowGraph]].
+ *
+ * @param identifier The identifier of this view within the graph.
+ * @param properties Properties of the view
+ * @param comment when defining a view
+ */
+case class PersistedView(
```

Comment:
Is it the same thing with the well-known "materialized view"?

Reply:
NV, I figured it out after reading the code, it represents a View registered in the catalog.